### PR TITLE
guard against truncated ciphertext in Aead::open (nss)

### LINKED ISSUE #30: integer underflow in open in nss/aead.rs
If `ClientResponse::decapsulate` is given an `enc_response` that doesn't actually have any ciphertext, `Aead::open` will cause integer underflow when determining inputs to `PK11_AEADOp`.

### PR DESCRIPTION
**Pull request by mozkeeler**
This fixes #30 by checking the input ciphertext length.

### CODE REVIEW

File: ohttp/src/nss/aead.rs [hunk 1]
```diff
@@ -149,6 +149,9 @@ impl Aead {
 
     pub fn open(&mut self, aad: &[u8], seq: SequenceNumber, ct: &[u8]) -> Res<Vec<u8>> {
         assert_eq!(self.mode, Mode::Decrypt);
+        if ct.len() < TAG_LEN {
```
Comment (martinthomson) [inline ohttp/src/nss/aead.rs:152]: You could also do:

```rust
let pt_expected = ct.len().checked_sub(TAG_LEN).ok_or(Error::Truncated)?;
```
with appropriate use of this value below.

**Review by martinthomson — APPROVED** [review]
A suggestion, but up to you.

Thanks for catching this one.

File: ohttp/src/nss/aead.rs (hunk 1 above)
Comment (mozkeeler) [reply ohttp/src/nss/aead.rs:152]: Oh I like that much better - thanks!

**Comment by mozkeeler** [comment]
Hi @martinthomson - is this good to go? If so, could you merge it, please? Thanks!

**Comment by martinthomson** [comment]
Sorry about that @mozkeeler, fixed.

**Pull request merged**